prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -5374,6 +5411,62 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
             .min()
             .unwrap()
     }
+
+    /// A local version of encode_place
+    fn encode_place(
+        &self,
+        place: &mir::Place<'tcx>,
+    ) -> EncodingResult<(vir::Expr, Vec<vir::Stmt>, ty::Ty<'tcx>, Option<usize>)> {
+        let (encoded_place, ty, variant_idx) = self.mir_encoder.encode_place(place)?;
+        // TODO: actual encoding of e.g. array access here
+        let (encoded_expr, encoding_stmts) = self.postprocess_place_encoding(encoded_place)?;
+
+        Ok((encoded_expr, encoding_stmts, ty, variant_idx))
+    }
+
+    fn encode_projection(
```

Comment:
Just looking at the code in this PR, it seems that this function is not needed.

Reply:
`encode_projection` is called in `ProcedureEncoder::encode_statement_at`, which has a `prusti_interface::environment::mir_utils::MirPlace` instead of a `rustc_middle::mir::Place`, so we can't call `encode_place` here, nor do i see how to recover/rebuild a `mir::Place` again from that.

Reply:
Isn't `ProcedureEncoder::encode_statement_at` calling `MirEncoder` with `self.mir_encoder.encode_projection`? I can't find a call of this `encode_projection`, but I might be missing something because I'm just doing textual search on the diff.

Reply:
uhm.. that should be [part of my changes](https://github.com/viperproject/prusti-dev/pull/470/files#diff-704bbc6b705baff9ae5a91e768c083923396f2cfd663646531528a9494384a23R1056). didn't show up when text-searching in the diff first, because the diff in `procedure_encoder.rs` was too big so github hid it by default :-/

Reply:
Yes, found it, sorry.